packages/core/aqua-to-js/src/index.ts

```diff
@@ -52,8 +53,7 @@ export default async function aquaToJs<T extends OutputType>(
         sources: generateSources(res, "js", packageJson),
         types: generateTypes(res, packageJson),
       }
-    : // TODO: probably there is a way to remove this type assert
-      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    : // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
You can avoid casting here if you just use overloads here as I suggested long ago when this function first appeared

```typescript
type JsOutput = {
  sources: string;
  types: string;
};

type TsOutput = {
  sources: string;
};

type NothingToGenerate = null;

export default async function aquaToJs(
  res: CompilationResult,
  outputType: "js",
): Promise<JsOutput | NothingToGenerate>;
export default async function aquaToJs(
  res: CompilationResult,
  outputType: "ts",
): Promise<TsOutput | NothingToGenerate>;

export default async function aquaToJs(
  res: CompilationResult,
  outputType: "js" | "ts",
): Promise<JsOutput | TsOutput | NothingToGenerate> {
  if (
    Object.keys(res.services).length === 0 &&
    Object.keys(res.functions).length === 0
  ) {
    return null;
  }

  const packageJson = await getPackageJsonContent();

  if (outputType === "js") {
    return {
      sources: generateSources(res, "js", packageJson),
      types: generateTypes(res, packageJson),
    };
  }

  return {
    sources: generateSources(res, "ts", packageJson),
  };
}
```

Reply:
nice